pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -383,15 +386,22 @@ private void ensureMinReplicas(TableConfig config) {
   @Path("/tables/{tableName}/rebalance")
   @ApiOperation(value = "Rebalances segments of a table across servers", notes = "Rebalances segments of a table across servers")
   public ZNRecord rebalance(
```

Comment:
Can we change this API to return both new idealstate mapping and partition assignment in a JSON format? This will be helpful for debugging since this API can potentially update ZK twice. We should be able to see both when we run with `dryrun` flag. For offline table, we can return the partition assignment as empty.

Reply:
Made this change